pkg/health/client/client.go

```diff
@@ -368,47 +388,45 @@ func FormatHealthStatusResponse(w io.Writer, sr *models.HealthStatusResponse, pr
 			localhost = node
 		}
 	}
-	if succinct {
-		fmt.Fprintf(w, "Cluster health:\t%d/%d reachable\t(%s)\n",
-			healthy, len(sr.Nodes), sr.Timestamp)
-		if printAll || healthy < len(sr.Nodes) {
-			fmt.Fprintf(w, "  Name\tIP\tNode\tEndpoints\n")
-		}
-	} else {
-		fmt.Fprintf(w, "Probe time:\t%s\n", sr.Timestamp)
-		fmt.Fprintf(w, "Nodes:\n")
-	}
+
+	fmt.Fprintf(w, "Cluster health:\t%d/%d reachable\t(%s)\n",
+		healthy, len(sr.Nodes), sr.Timestamp)
 
 	if localhost != nil {
-		formatNodeStatus(w, localhost, printAll, succinct, verbose, true)
-		maxLines--
+		printed := formatNodeStatus(w, localhost, allNodes, verbose, true)
+		if printed {
+			printedLines++
+		}
 	}
 
 	nodes := sr.Nodes
 	sort.Slice(nodes, func(i, j int) bool {
 		return strings.Compare(nodes[i].Name, nodes[j].Name) < 0
 	})
-	for n, node := range nodes {
-		if maxLines > 0 && n > maxLines {
+	for _, node := range nodes {
+		if printedLines == maxLines {
 			break
 		}
 		if node == localhost {
 			continue
 		}
-		formatNodeStatus(w, node, printAll, succinct, verbose, false)
+		printed := formatNodeStatus(w, node, allNodes, verbose, false)
+		if printed {
+			printedLines++
+		}
 	}
-	if maxLines > 0 && len(sr.Nodes)-healthy > maxLines {
-		fmt.Fprintf(w, "  ...")
+	if len(sr.Nodes)-printedLines-healthy > 0 {
+		fmt.Fprintf(w, "  ...\n")
 	}
 }
 
 // GetAndFormatHealthStatus fetches the health status from the cilium-health
 // daemon via the default channel and formats its output as a string to the
 // writer.
 //
-// 'succinct', 'verbose' and 'maxLines' are handled the same as in
+// 'verbose' and 'maxLines' are handled the same as in
 // FormatHealthStatusResponse().
-func GetAndFormatHealthStatus(w io.Writer, succinct, verbose bool, maxLines int) {
+func GetAndFormatHealthStatus(w io.Writer, allNodes bool, verbose bool, maxLines int) {
```

Comment:
Prefer allNodes, verbose bool

Reply:
Yes, I didn't know that Go has this kind of grouping for data types. Thank you.